pulsebot/pulse_dispatch.py

```diff
@@ -231,51 +190,65 @@ def get_one():
                 continue
 
             try:
-                comments = "\n".join(self.bugzilla.get_comments(info.bug))
+                comments = self.bugzilla.get_comments(info.bug)
+                comments_string = ""
+                for c in comments:
+                    comments_string += c.get("text", "") + "\n"
```

Comment:
You build the `comments_string` but only use it to check if `url[-12:] not in comments_string` down the line. Instead you should keep `comments` as a list and then change the check to be `if not any(url[-12:] in comment for comment in comments):`.